appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Stichting Yona Foundation
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v.2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server
+
+import groovy.json.*
+
+class ActivityTest extends AbstractAppServiceIntegrationTest
+{
+	def 'Get days activity'()
+	{
+		given:
+		def richard = addRichard()
+		analysisService.postToAnalysisEngine(richard, ["Gambling"], "http://www.poker.com")
+
+		when:
+		def response = appService.getDayActivityOverviews(richard)
+
+		then:
+		response.status == 200
+		response.responseData._embedded
+		response.responseData._embedded."yona:dayActivities"
+		response.responseData._embedded."yona:dayActivities".size() == 1
+		//response.responseData._embedded."yona:dayActivities"[0].spread
+		response.responseData._embedded."yona:dayActivities"[0].totalActivityDurationMinutes == 0
+		response.responseData._embedded."yona:dayActivities"[0].goalAccomplished == false
```

Comment:
This is weird. The duration beyond goal is zero, but the goal is still not accomplished. Technically, this is understandable, but functionally, it's wrong. I propose we fix this in the analysis engine. If we get the first web event, we set the end time one second past the start time. And when we round seconds to minutes, we ensure we don't round to zero.

With that, the story is functionally correct: the user did something outside the goal limits for a small period of time. Technically, we can now do away wit the ``goalAccomplished`` flag, as ``totalMinutesBeyondGoal`` already communicates the status.